client/src/layout/Scroller.js

```diff
@@ -0,0 +1,45 @@
+import React, { useContext, useEffect } from "react";
+import InfiniteScroll from "react-infinite-scroller";
+import { UserContext } from "../context/user";
+import { getMentions } from "../hooks/getMentions";
+import Spinner from "./Spinner";
+import Mention from "../layout/Mention";
+
+const Scroller = ({
+  sort,
+  error,
+  loadmore,
+  hasMore,
+  setHasMore,
+  mentionDatas,
+  setMentionDatas,
+}) => {
+  const { dispatch, searchTerm, user } = useContext(UserContext);
+
+  useEffect(() => {
+    getMentions(dispatch, searchTerm, user.platforms, 1, sort)
+      .then((data) => {
+        setMentionDatas(data.mentions);
+        setHasMore(data.nextPage ? true : false);
+      })
+      .catch((err) => alert("Cookie expired. Please log in again"));
+  }, [sort]);
```

Comment:
Just curious, should searchTerm and user.platforms also be here too?

Reply:
Those are parameters expected by the GetMentions hook. I suppose the code would break without them, not 100% sure though. Also, we want to scroll only through mentions matching the searchTerm and platforms requested by user.